src/subs/Match.cpp

```diff
@@ -186,18 +186,148 @@ Match::Match(unique_ptr<Detection> e)
           isSolved(false),
           trivialWeights(false) {
     add(std::move(e));
+    // Set null field projection.
+    astrometry::Orientation orient(astrometry::SphericalICRS(0, 0));
+    fieldProjection = new astrometry::Gnomonic(orient);
+
+}
+
+Match::Match(std::vector<double> xw,
+             std::vector<double> yw,
+             std::vector<double> xCov,
+             std::vector<double> yCov,
+             std::vector<double> xyCov,
+             std::vector<double> mjd,
+             vector<std::vector<double>> observatory,
+             double fieldRA,
+             double fieldDec,
+             double fieldMJD)
+    : elist(),
+      nFit(0),
+      dof(0),
+      isReserved(false),
+      isPrepared(false),
+      isMappedFit(false),
+      isMappedAll(true),
+      isSolved(false),
+      trivialWeights(false)
+{
+
+    if (not (xw.size() == yw.size()) && (xw.size() == xCov.size()) && (xw.size() == yCov.size()) 
+        && (xw.size() == xyCov.size()) && (xw.size() == mjd.size()) && (xw.size() == observatory.size())) {
+        throw std::runtime_error("Input arrays must have same length.");
+    }
+    astrometry::Orientation orient(
+        astrometry::SphericalICRS(std::move(fieldRA) * WCS_UNIT,
+                                    std::move(fieldDec) * WCS_UNIT));
+    fieldProjection = new astrometry::Gnomonic(orient);
```

Comment:
```suggestion
    fieldProjection.reset(new astrometry::Gnomonic(orient));
```
(previous comment also applies)